lib/embedded_ansible.rb

```diff
@@ -43,9 +43,45 @@ def api_connection_raw(host, port)
     )
   end
 
+  def find_or_create_secret_key
+    miq_database.ansible_secret_key || miq_database.ansible_secret_key = SecureRandom.hex(16)
+  end
+
+  def find_or_create_admin_authentication
+    miq_database.ansible_admin_authentication || miq_database.set_ansible_admin_authentication(:password => generate_password)
+  end
+
+  def find_or_create_rabbitmq_authentication
+    miq_database.ansible_rabbitmq_authentication || miq_database.set_ansible_rabbitmq_authentication(:password => generate_password)
+  end
+
+  def find_or_create_database_authentication
+    auth = miq_database.ansible_database_authentication
+    return auth if auth
+
+    auth = miq_database.set_ansible_database_authentication(:password => generate_password)
+
+    database_connection.select_value("CREATE ROLE #{database_connection.quote_column_name(auth.userid)} WITH LOGIN PASSWORD #{database_connection.quote(auth.password)}")
+    database_connection.select_value("CREATE DATABASE awx OWNER #{database_connection.quote_column_name(auth.userid)} ENCODING 'utf8'")
+
+    auth
+  end
+
+  def generate_password
```

Comment:
I can't tell if they aren't already, but these should probably be private methods from here down.

Reply:
Yup, they're private starting at [line 34](https://github.com/ManageIQ/manageiq/pull/16205/files#diff-a43db89587c394a13dcf99450055887aR34)